cvmfs/catalog.h

```diff
@@ -163,7 +163,7 @@ class Catalog : public SingleCopy {
   void TakeDatabaseFileOwnership();
   void DropDatabaseFileOwnership();
   bool OwnsDatabaseFile() const {
-    return NULL != database_ && database_->OwnsFile();
+    return ((database_ != NULL) && database_->OwnsFile()) || managed_database_;
```

Comment:
mmh, double dispatch might work... I would vote for revisiting the `Catalog` resp. `CatalogManager` facilities on a more basic level though. Not sure what that means yet, but they are not unit tested and hence need some care anyways.